### PR TITLE
boost-python3: Build a bottle for Linuxbrew

### PR DESCRIPTION
**Pull request by akaitake**
- [x] Have you followed the [guidelines for contributing](https://github.com/Linuxbrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Linuxbrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/boost-python3.rb [hunk 1]
```diff
@@ -45,7 +45,8 @@ def install
     inreplace "bootstrap.sh", "using python", "#using python"
 
     pyver = Language::Python.major_minor_version "python3"
-    py_prefix = Formula["python3"].opt_frameworks/"Python.framework/Versions/#{pyver}"
+    # py_prefix = Formula["python3"].opt_frameworks/"Python.framework/Versions/#{pyver}"
+    py_prefix = `python3 -c "import sys; print(sys.prefix)"`.strip
```
Comment (iMichka) [inline Formula/boost-python3.rb]: Don't leave commented out code; remove that line and rewrite this as:
```
if OS.mac?
  py_prefix = Formula["python3"].opt_frameworks/"Python.framework/Versions/#{pyver}"
else
  py_prefix = `python3 -c "import sys; print(sys.prefix)"`.strip
end
```

**Comment by akaitake** [comment]
Thank you for your comment.
Yes, I can build this in my local environment. So I can not understand why circle.ci failed.

File: Formula/boost-python3.rb (hunk 1 above)
Comment (denis-sumin) [reply Formula/boost-python3.rb]: This way of getting the path doesn't work for me; the obtained prefix is `/home/linuxbrew/.linuxbrew/opt/python3`.

**Comment by denis-sumin** [comment]
On my system, the installation with the edits proposed succeeds but it doesn't build `libboost_numpy3`. The log of b2 just doesn't mention any trial to build numpy lib. Perhaps, it's because something is wrong with the numpy local installation. Trying to solve it anyhow, with no luck still.

**Comment by denis-sumin** [comment]
The following workaround worked for me:
* use `--env=std`
* hardcode the current python path as `py_prefix`
* comment out the numpy building step (since I have it installed as a package); it will fail to build with the `--env=std`

**Pull request closed without merging**